src/chunk.js

```diff
@@ -404,6 +407,13 @@ class Chunks {
    * @private
    */
   executeAsap_(idleDeadline) {
+    // If we've spent over 5 millseconds executing the
+    // last instruction yeild back to the main thread.
+    // 5 milliseconds is a magic number.
+    if (Date.now() - this.timeSinceLastExecution_ > 5) {
```

Comment:
We need to experiment-guard this change. That should also make the merge trivial for now, because the existing unit tests don't turn it on.

One question: Did you try this locally and see whether it

- works as intended in the perf timeline
- and whether it leads to over-painting (paint of partially initialized page when we yield)

?

In any case, with an experiment guard we can try it out on many real world pages.

Reply:
Tested this change locally with simulated networking and CPU throttling. The impact was a reduction in FID by ~15% on the worst case documents.

Adding an experiment now.